src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Protocol/SemanticTokensEncoder.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Texl.Intellisense;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol.Protocol
+{
+    /// <summary>
+    /// An entity with methods to encode the tokens for the responses to the requests for semantic tokens in language server.
+    /// </summary>
+    internal static class SemanticTokensEncoder
+    {
+        /// <summary>
+        /// There are no token modifiers in powerfx.
+        /// </summary>
+        private const uint NoTokenModifiersMask = 0u;
+
+        /// <summary>
+        /// A constant to indicate how many slots each token would occupy in the encoded tokens collection.
+        /// </summary>
+        internal const int SlotsPerToken = 5;
+
+        /// <summary>
+        /// A mapping from token type to its unsigned interger encoded value. Each token type is encoded using its position in the TokenType enum.
+        /// </summary>
+        private static readonly IDictionary<TokenType, uint> EncodedTokenTypes = Enum.GetValues(typeof(TokenType))
```

Comment:
IReadOnlyDictionary?

Reply:
Wasn't aware of the existence of this interface

Reply:
Thank you for pointing this out. IReadOnlyDictionary make sense